analyzer/analyzer.go

```diff
@@ -44,12 +62,31 @@ func New() *analysis.Analyzer {
 		Run:      n.run,
 		Requires: []*analysis.Analyzer{inspect.Analyzer},
 	}
-	r.Flags.BoolVar(&n.intConv, "int-conversion", true, "optimizes even if it requires an int or uint type cast")
-	r.Flags.BoolVar(&n.errError, "err-error", false, "optimizes into err.Error() even if it is only equivalent for non-nil errors")
-	r.Flags.BoolVar(&n.errorf, "errorf", true, "optimizes fmt.Errorf")
-	r.Flags.BoolVar(&n.sprintf1, "sprintf1", true, "optimizes fmt.Sprintf with only one argument")
+	r.Flags.BoolVar(&n.intFormat.enabled, "integer-format", true, "enable/disable optimization of integer formatting")
+	r.Flags.BoolVar(&n.intFormat.intConv, "int-conversion", true, "optimizes even if it requires an int or uint type cast")
+	r.Flags.BoolVar(&n.errFormat.enabled, "error-format", true, "enable/disable optimization of error formatting")
+	r.Flags.BoolVar(&n.errFormat.errError, "err-error", false, "optimizes into err.Error() even if it is only equivalent for non-nil errors")
+	r.Flags.BoolVar(&n.errFormat.errorf, "errorf", true, "optimizes fmt.Errorf")
+	r.Flags.BoolVar(&n.boolFormat, "bool-format", true, "enable/disable optimization of bool formatting")
+	r.Flags.BoolVar(&n.hexFormat, "hex-format", true, "enable/disable optimization of hex formatting")
+	r.Flags.BoolVar(&n.strFormat.enabled, "string-format", true, "enable/disable optimization of string formatting")
+	r.Flags.BoolVar(&n.strFormat.sprintf1, "sprintf1", true, "optimizes fmt.Sprintf with only one argument")
+	r.Flags.BoolVar(&n.strFormat.strconcat, "strconcat", true, "optimizes into strings concatenation")
 	r.Flags.BoolVar(&n.fiximports, "fiximports", true, "fix needed imports from other fixes")
-	r.Flags.BoolVar(&n.strconcat, "strconcat", true, "optimizes into strings concatenation")
+
+	if !n.intFormat.enabled {
+		n.intFormat.intConv = false
+	}
+
+	if !n.errFormat.enabled {
+		n.errFormat.errError = false
+		n.errFormat.errorf = false
+	}
+	if !n.strFormat.enabled {
+		n.strFormat.sprintf1 = false
+		n.strFormat.strconcat = false
+	}
```

Comment:
So here, it can be [simplified](https://github.com/catenacyber/perfsprint/pull/41#discussion_r1935362061) and clearer


```suggestion
	if n.intFormat.deactivated {
		n.intFormat.intConv = false
	}

	if n.errFormat.deactivated {
		n.errFormat.errError = false
		n.errFormat.errorf = false
	}

	if n.strFormat.deactivated {
		n.strFormat.sprintf1 = false
		n.strFormat.strconcat = false
	}
```

Reply:
I do not find this simpler nor more complex